### PR TITLE
Update to actual versions of cfssl and go itself

### PR DESCRIPTION
**Pull request by Envek**
When I tried to update https://github.com/gravitational/teleport/pull/35652 with changes from https://github.com/gravitational/license/pull/10 I've got a bunch of following errors:

```
github.com/gravitational/license@v0.0.0-20231220172702-b016eb344a22/authority/ca.go:148:20: cannot use existingKey (variable of type *ExistingKey) as *csr.KeyRequest value in assignment
github.com/gravitational/license@v0.0.0-20231220172702-b016eb344a22/authority/ca.go:150:25: undefined: csr.BasicKeyRequest
```

Seems that [cfssl](https://github.com/cloudflare/cfssl) has got a lot of updates since this library was written and older versions can't be used in Teleport `go.mod` (cfssl was forcibly updated in Teleport's `go.mod` when I executed `go mod tidy`).

So I decided to refresh this repo a bit to use up-to-date cfssl API, introduce go.mod to lock actual cfssl version, and add a test run to GitHub Actions.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    name: "Go ${{ matrix.go }}"
+    strategy:
+      matrix:
+        include:
+          - go: '1.21.x'
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
```
Comment (zmb3) [inline .github/workflows/test.yml:15]:
```suggestion
        uses: actions/setup-go@v5
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        with:
+          go-version: '1.21.x'
+      - name: Install dependencies
+        run: go get .
```
Comment (zmb3) [inline .github/workflows/test.yml:19]:
```suggestion
```

Explicit `go get` is not necessary.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      - name: Test
+        run: go test -v ./...
```
Comment (zmb3) [inline .github/workflows/test.yml:21]:
```suggestion
        run: go test -v -race ./...
```

**Review by zmb3 — APPROVED** [review]

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    strategy:
+      matrix:
+        include:
+          - go: '1.21.x'
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.21.x'
```
Comment (reedloden) [inline .github/workflows/test.yml:17]:
```suggestion
          go-version-file: 'go.mod'
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: "Go ${{ matrix.go }}"
+    strategy:
+      matrix:
+        include:
+          - go: '1.21.x'
```
Comment (reedloden) [inline .github/workflows/test.yml:11]:
```suggestion
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+name: Tests
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: "Go ${{ matrix.go }}"
```
Comment (reedloden) [inline .github/workflows/test.yml:7]:
```suggestion
    name: "Tests"
```

File: .github/workflows/test.yml (hunk 4 above)
Comment (zmb3) [reply .github/workflows/test.yml:17]: This is subjective. The version in `go.mod` is the minimum required Go version.

I typically argue that go.mod should aim to keep that version as _low_ as possible (to maintain compatibility), and the version used in CI should stay as _high_ as possible (to protect against regressions and vulnerabilities).

Comment (Envek) [reply .github/workflows/test.yml:17]: Is License lib supposed to be used with different versions of Go?

File: .github/workflows/test.yml (hunk 6 above)
Comment (Envek) [reply .github/workflows/test.yml:7]: That duplicates workflow-level `Tests` name and will appear in UI as `Tests / Tests` which is a bit confusing. IMHO it is better to use command name for clarity, e.g. `Tests / go test`
```suggestion
    name: "go test"
```

File: .github/workflows/test.yml (hunk 4 above)
Comment (zmb3) [reply .github/workflows/test.yml]: Not necessarily, but there's no need to artificially constrain compatibility.

File: .github/workflows/test.yml (hunk 6 above)
Comment (zmb3) [reply .github/workflows/test.yml]: I'm fine with `go test`

**Review by zmb3 — APPROVED** [review]

File: .github/workflows/test.yml (hunk 4 above)
Comment (strideynet) [reply .github/workflows/test.yml]: I'm with Zac here - CI should be using the most recent version of Go toolchain possible.

**Review by strideynet — APPROVED** [review]

**Comment by Envek** [comment]
Updated packages to get rid of stale dependency warning (wondering why old packages were picked up when I generated `go.mod` initially)

**Comment by zmb3** [comment]
@Envek you can stop dismissing reviews every time you make a change. It means I have to continuously chase down reviewers for you in order to get this tiny PR merged.

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    name: "go test"
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
```
Comment (zmb3) [inline .github/workflows/test.yml:13]:
```suggestion
          go-version: 'stable'
```

This repo is infrequently updated, so let's just configure it to always run CI with the latest stable version of Go.

File: go.mod [hunk 8]
```diff
@@ -0,0 +1,29 @@
+module github.com/gravitational/license
+
+go 1.21
```
Comment (zmb3) [inline go.mod:3]:
```suggestion
go 1.19
```

Teleport 12 is still specifying Go 1.19 as a minimum, and we don't use any newer syntax or language features in Go 1.21 so there's no need to break support.

**Review by zmb3 — APPROVED** [review]

**Comment by Envek** [comment]
> you can stop dismissing reviews every time you make a change.

That's not me. GitHub automatically dismisses reviews on any change.

**Review by strideynet — APPROVED** [review]

**Pull request merged**